journals/apps/journals/models.py

```diff
@@ -596,3 +596,19 @@ def get_nested_children(self):
 
         structure["children"] = [child.specific.get_nested_children() for child in children]
         return structure
+
+
+class UserPageVisit(models.Model):
+    """
+    Record the datetime at which user visited the page.
+    """
+    user = models.ForeignKey(User, null=True, on_delete=models.SET_NULL)
+    page = models.ForeignKey('wagtailcore.Page', null=True, on_delete=models.SET_NULL)
+    visited_at = models.DateTimeField(auto_now=True)
+    stale = models.BooleanField(
```

Comment:
Stale field is not needed here.

Reply:
I think @tuchfarber requested we add that stale field. See comment in the story